packages/map-template/src/components/QRCodeDialog/QRCodeDialog.jsx

```diff
@@ -0,0 +1,95 @@
+import React, { useEffect, useRef } from "react";
+import { useRecoilState, useRecoilValue } from 'recoil';
+import './QRCodeDialog.scss';
+import showQRCodeDialogState from "../../atoms/showQRCodeDialogState";
+import primaryColorState from "../../atoms/primaryColorState";
+import apiKeyState from "../../atoms/apiKeyState";
+import currentLocationState from "../../atoms/currentLocationState";
+import QRCode from 'qrcode';
+import kioskLocationState from "../../atoms/kioskLocationState";
+import logoState from "../../atoms/logoState";
+import mapboxAccessTokenState from "../../atoms/mapboxAccessTokenState";
+import gmApiKeyState from "../../atoms/gmApiKeyState";
+
+/**
+ * Handle the QR Code dialog.
+ *
+ */
+function QRCodeDialog() {
+    const [, setShowQRCodeDialog] = useRecoilState(showQRCodeDialogState);
+    const elementRef = useRef()
+
+    const primaryColorProp = useRecoilValue(primaryColorState);
+    const apiKeyProp = useRecoilValue(apiKeyState);
+    const logoProp = useRecoilValue(logoState);
+    const gmApiKeyProp = useRecoilValue(gmApiKeyState);
+    const mapboxAccessTokenProp = useRecoilValue(mapboxAccessTokenState);
+
+    const directionsFrom = useRecoilValue(kioskLocationState);
+    const directionsTo = useRecoilValue(currentLocationState);
+
+    useEffect(() => {
+        if (directionsFrom && directionsTo) {
+            // The target URL when the user opens the QR code dialog
+            let targetUrl = window.location.origin;
+
+            // The interface for the existing URL search params
+            const currentParams = new URLSearchParams(window.location.search);
```

Comment:
Just came to mind: Should we check for supportsUrlParameters here? We risk clashing with similarly named query parameters in case the customer has their own.

Reply:
Hmmm, sure, we could check for that prop here, but by default we assume that the `supportsUrlParameters` is true. So that would mitigate the risk of clashing in case we the customer user **props only** and has `supportsUrlParameters` set to false.
But if the customer has something like `primaryColor` or `logo`, and if we have `supportsUrlParameters` set to true, it would would clash anyway, right? @ammapspeople

Reply:
Imagine the case where

1. The customer has `supportsUrlParameters` to false (which is the default)
2. Have their own URL parameter scheme in place. Some of these are named `logo` and `primaryColor`

What will happen to the QR code URL?

Reply:
That's a good point 💡 I will add a check for `supportsUrlParameters`

Reply:
Ready for review again 😄 @ammapspeople

Reply:
@andreeaceachir142 I maintain my approval 😄

Reply:
🎉